fix(web): Debug appendTrailingSlash; simplify React route

RyanSkonnord opened the pull request:
~~Apologies for the elaborate kludge.~~ Background:
* https://github.com/getsentry/sentry/pull/25470
* https://getsentry.atlassian.net/browse/ER-587
* https://getsentry.atlassian.net/browse/ER-663

**Update:** No longer a kludge! 🎉

RyanSkonnord committed:
fix(web): Redirect additional specific paths at org slug level

markstory reviewed (approved):
Seems reasonable to me.

JoshFerge reviewed (commented):
does this run into the same issue as before with the failed acceptance test b.c. of our react router removing the query string?

RyanSkonnord committed:
Remove releases

RyanSkonnord commented:
> does this run into the same issue as before with the failed acceptance test b.c. of our react router removing the query string?

That is correct. The integration test fails whenever it catches a `*/releases/*` path -- that is, the same problem with the `[\w-]+` solution, which I inadvertently reproduced by dropping "releases" into the list.

In order for this "bandaid" solution to proceed, I'd want to do some combination of:
1. forego covering "releases" in the fix;
2. do a quick pass to ensure that it won't break other addresses with query strings;
3. fix the deferred bug in the React routing.

I figure "(1 AND 2) OR 3" is necessary.

RyanSkonnord committed:
fix(web): Debug appendTrailingSlash; simplify React route

Fix the appendTrailingSlash function on a React route so that it no
longer removes the search and hash parts of the rewritten location.

Simplify an associated URL-matching pattern that no longer needs to
dodge that bug. This has to do with a bug that (while not affecting
logged-in users) would cause logged-out users not to be redirected from
certain org-specific addresses to the correct org-branded login page.
The fix for that bug should now be generalized to all org pages.

RyanSkonnord committed:
Merge branch 'master' into redirect-org-level-pages

RyanSkonnord commented:
@JoshFerge All right, the underlying problem is now fixed for realsies. Care to re-approve?

JoshFerge reviewed (approved):
🔥

markstory commented on static/app/routes.tsx:
```diff
@@ -34,7 +34,8 @@ import SettingsWrapper from 'app/views/settings/components/settingsWrapper';
   if (lastChar !== '/') {
-    replace(nextState.location.pathname + '/');
+    const pathname = nextState.location.pathname + '/';
+    replace(pathname + nextState.location.search + nextState.location.hash);
```
:clap:

RyanSkonnord committed:
Merge branch 'master' into redirect-org-level-pages

RyanSkonnord committed:
Merge branch 'master' into redirect-org-level-pages
